src/adapters/MeetingsJSONAdapter.js

```diff
@@ -432,24 +446,23 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
    */
   async handleLocalShare(ID) {
     await this.updateMeeting(ID, async (meeting) => {
-      const updatedMeeting = {...meeting};
+      const updates = {
+        localShare: meeting.localShare,
+        remoteShare: meeting.remoteShare,
+      };
 
       if (meeting.localShare) {
-        updatedMeeting.localShare.getTracks()[0].stop();
-        updatedMeeting.localShare = null;
+        updates.localShare.getTracks()[0].stop();
+        updates.localShare = null;
       } else {
-        updatedMeeting.localShare = await this.getDisplayStream();
-        updatedMeeting.remoteShare = null;
-
-        if (meeting.localShare) {
-          // Handle browser's built-in stop Button
-          updatedMeeting.localShare.getVideoTracks()[0].onended = () => {
-            updatedMeeting.localShare = null;
-          };
-        }
+        updates.localShare = await this.getDisplayStream();
+        updates.localShare.getVideoTracks()[0].onended = () => {
+          updates.localShare = null;
```

Comment:
this is a callback that is called much later, when the user presses the default browser "Stop sharing your screen" button. The modification done to `updates` at that point won't update the meeting object. I would recommend calling the `handleLocalShare` in this callback to clean up the meeting object:
```js
updates.localShare.getVideoTracks()[0].onended = () => handleLocalShare(ID);
```